frame/staking/src/tests.rs

```diff
@@ -440,13 +440,24 @@ fn no_candidate_emergency_condition() {
 			let res = Staking::chill(Origin::signed(10));
 			assert_ok!(res);
 
-			// trigger era
-			mock::start_active_era(1);
+			let current_era = CurrentEra::get();
 
-			// Previous ones are elected. chill is invalidates. TODO: #2494
+			// try trigger new era
+			mock::run_to_block(20);
+			assert_eq!(
+				*staking_events().last().unwrap(),
+				RawEvent::StakingElectionFailed,
+			);
+			// No new era is created
+			assert_eq!(current_era, CurrentEra::get());
+
+			// Go to far further session to see if validator have changed
+			mock::run_to_block(100);
+
+			// Previous ones are elected. chill is not effective in active era (as era hasn't changed)
```

Comment:
Can we assert the era has not changed?

Also, does the status of era forcing change? (From  my understanding it does not change if the election fails, but wanted to check)

Reply:
Yes I added a check in this commit https://github.com/paritytech/substrate/pull/8912/commits/86dab55c65fc90d2ddd040a6a6a3dbd93807c7e6

For the forcing changing I also test it is reset to default only after a successfully planned era.